src/test/scala/ciris/aws/ssm/SsmAsyncClientMock.scala

```diff
@@ -0,0 +1,45 @@
+package ciris.aws.ssm
+
+import software.amazon.awssdk.services.ssm.SsmAsyncClient
+import software.amazon.awssdk.services.ssm.model.{
+  GetParameterRequest,
+  GetParameterResponse,
+  Parameter,
+  ParameterNotFoundException
+}
+
+import java.util.concurrent.CompletableFuture
+import scala.collection.mutable
+
+case class SsmAsyncClientMock(initialValues: (String, String)*) extends SsmAsyncClient {
+
+  def serviceName(): String = "mock-ssm"
+
+  def close(): Unit = ()
+
+  private[this] val secrets = mutable.Map(initialValues: _*)
```

Comment:
not clear why we need mutability here?

Reply:
Because I thought I might need to mutate the fake params, but I don't - fixed. :)